stack.go

```diff
@@ -163,6 +164,19 @@ func longFuncName(name string) string {
 func CaptureStackTrace(skip int) StackTrace {
 	frames := make([]uintptr, 100)
 	length := runtime.Callers(skip+2, frames[:])
+
+	if init := initializing(); init {
```

Comment:
should we use compare and swap instead here?

Reply:
We can't, because we must first verify that we've completed the init phase of Go programs before we can set the `intialized` variable to `1`.